doc/adr/0004-git-branching-strategy.md

```diff
@@ -0,0 +1,86 @@
+developed code. There will be a 1 to 1 relationship between deployed
+environments and repository brnaches. The only exception to this 1 to 1
+relationship will be the _**master**_ branch. The _**master**_ branch may be used to
+back any environment that needs to be live like.
+
+Each branch name will be broken down into 3 components. An optional category,
+default-environment-reference reference and a brief summary. The approved format
+is shown below.
+
+  `<category>/<default-environment-reference>-<summary>`
+  OR
+  `<default-environment-reference>-<summary>`
+
+each component has a set of rules it must adhere to.
+
+  If present _**category**_ MUST be one of _**feature**_, _**bugfix**_, _**security**_ or _**personal**_.
+
+  each _**category**_ describes the type of work to be found in that branch.
+
+  - _**feature**_ branches introduce some new functionality to the repository.
+  - _**bugfix**_ branches are fixing some issue with existing code.
+  - _**security**_ branches are resolving a security issue with the code
+  - _**personal**_ branches are for work that is independant of a specific ticket.
+
+    Determination of which category to use is fairly arbitrary and engineers
+    judgement should be used. There is likely to be some overlap, so just use
+    whichever fits best or no _**category**_. The main exception is for the  _**personal**_ category
+    which is to be used for experimenting with different approaches, but with
+    no view to deployment. work from a personal branch would then be pulled
+    into a branch in one of the standard categories.
+
+  _**default-environment-reference**_ is to be used to namespace environments created using
+  the branch. This should only be a default and could be overridden by an
+  explicit decision. For branches in _**feature**_, _**bugfix**_ and _**security**_
+  categories the _**default-environment-reference**_ MUST be the jira ticket id.
+  This allows details about the work contained in the branch to be quickly and
+  easily correlated with work in Jira.
+
+  For branches in the _**personal**_ category the _**default-environment-reference**_
+  should maintain the same format as a jira ticket reference but using an
+  engineers username instead of jira project. E.G. pmccabe-1
+
+  _**summary**_ should be a consise description of the work contained in the branch.
+  This should consist only of lowercase letters numbers and dashes to
+  seperate words. This should not be a replication of the summary line in the
+  jira ticket. but should be more consise.
+
+  #### Examples:
+
+  - feature/JIRA-34-create-new-weblogic-servers
+  - bugfix/JIRA-35-enable-selinux
+  - security/JIRA-36-update-to-latest-rhel-version
+  - personal/PMCCABE-1-experiment-with-vmextension-script-deployments
+
+Each branch should be created from `origin/master` then worked on and tested.
+When ready a Pull Request should be generated to merge back into `origin/master`
```

Comment:
:+1: too _should_. We'll rarely want to branch of branches, but I have found that useful in the past to provide stepped reviews when there's not been a reviewer for a day or so rather than one big wall of code.